web/src/screens/filters/list.tsx

```diff
@@ -141,6 +144,28 @@ export default function Filters({}: FilterProps){
   };
   
   const [showDropdown, setShowDropdown] = useState(false);
+  const dropdownRef = useRef(null);
+  const addFilterRef = useRef(null);
+
+  useEffect(() => {
+    function handleClickOutside(event: { target: any }) {
+      if (
+        dropdownRef.current &&
+        !(dropdownRef.current as any).contains(event.target) &&
+        !(addFilterRef.current as any).contains(event.target)
+      ) {
+        setShowDropdown(false);
+      }
+    }
+
+    document.addEventListener("click", handleClickOutside);
+
+    return () => {
+      document.removeEventListener("click", handleClickOutside);
+    };
+  }, [dropdownRef, addFilterRef]);
```

Comment:
Pretty sure you don't have to add refs as dependencies in useEffect. If you're relying on VSCode warnings/hints, I'll tell you that the useEffect plugin is lying a good amount of time. Might be wrong in this case however.